File: myapp/static/js/presenters/inspectionCardPlansPresenter.js
// static/js/presenters/inspectionCardPlansPresenter.js
import { fmtDt, fmtText, fmtManHours, fmtPractitioners } from './formatters.js';

export function buildInspectionCardPlansTableVM(res, { title } = {}) {
  const plans = res?.plans ?? []; // サーバpayloadに合わせて
  return {
    title: fmtText(title ?? '履歴'),
    columns: [
      { key: 'p_date', label: '計画日' },
      { key: 'implementation_date', label: '実施日' },
      { key: 'status', label: 'ステータス' },
      { key: 'result', label: '結果' },
      { key: 'points_to_note', label: '指摘事項' },
      { key: 'practitioners', label: '実施者' },
      { key: 'result_man_hours', label: '実施工数' },
    ],
    rows: plans.map(p => ({
      plan_id: p?.plan_id,
      p_date: fmtText(p?.p_date?.h_date),                 // dateはAPI側でstring化してるならfmtTextでOK
      implementation_date: fmtDt(p?.implementation_date),
      status: fmtText(p?.status),
      result: fmtText(p?.result),
      points_to_note: fmtText(p?.points_to_note || p?.comment),
      practitioners: fmtPractitioners(p?.practitioners),
      result_man_hours: fmtManHours(p?.result_man_hours),
    })),
  };
}

File: myapp/presenters/inspection_card_plans_presenter.py
from __future__ import annotations
from typing import Any, Dict, List

from myapp.presenters.formatters import as_text, as_int, dt_iso, d_iso, member_brief, safe_get

def _plan_history_row(plan) -> dict:
    p_date = getattr(plan, "p_date", None)

    practitioners_rows: List[dict] = []
    for pr in getattr(plan, "practitioners", []).all():
        practitioners_rows.append({
            "id": as_int(getattr(pr, "id", 0)),
            "member": member_brief(getattr(pr, "member_id", None)),
        })

    return {
        "plan_id": as_int(getattr(plan, "plan_id", 0)),
        "status": as_text(getattr(plan, "status", "")),
        "result": as_text(getattr(plan, "result", "")),
        "points_to_note": as_text(getattr(plan, "points_to_note", "")),
        "result_man_hours": as_int(getattr(plan, "result_man_hours", 0)),
        "implementation_date": dt_iso(getattr(plan, "implementation_date", None)),
        "comment": as_text(getattr(plan, "comment", "")),
        "p_date": {
            "h_date": d_iso(safe_get(p_date, "h_date", None)),
            "date_alias": as_text(safe_get(p_date, "date_alias", "")),
        },
        "practitioners": practitioners_rows,
    }

def build_inspection_card_plans_payload(*, inspection_no: str, plans: list) -> Dict[str, Any]:
    return {
        "status": "success",
        "inspection_no": as_text(inspection_no),
        "plans": [_plan_history_row(p) for p in plans],
    }